# Task 22 — Email Domain Classification

## Business Question

How many customers use Gmail, Yahoo, and Outlook email providers?

## Analysis

Customers were segmented by mail domains:
- gmail
- yahoo
- outlook

## Code

In [1]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
conditions = [
    starlink["email"].str.lower().str.contains("gmail"),
    starlink["email"].str.lower().str.contains("yahoo"),
    starlink["email"].str.lower().str.contains("outlook"),
]
choices = [
    "Gmail",
    "Yahoo",
    "Outlook"
]
starlink["email_provider"] = np.select(
    conditions,
    choices,
    default = "Other"
)
print(starlink["email_provider"].value_counts())

email_provider
Other      5687
Yahoo      1456
Outlook    1431
Gmail      1427
Name: count, dtype: int64


### Result

Gmail users — 1,427
Outlook users — 1,431
Yahoo users — 1,456

### Business Insight

Most customers use email providers outside Gmail, Yahoo, and Outlook, accounting for 5,687 customers. Among the identified major providers, Yahoo, Outlook, and Gmail have relatively similar numbers of users.